apps/web/src/app/(unauthenticated)/unverified-account/page.tsx

```diff
@@ -0,0 +1,68 @@
+'use client';
+
+import { useState } from 'react';
+
+import { useSearchParams } from 'next/navigation';
+
+import { Mails } from 'lucide-react';
+
+import { ONE_SECOND } from '@documenso/lib/constants/time';
+import { trpc } from '@documenso/trpc/react';
+import { Button } from '@documenso/ui/primitives/button';
+import { useToast } from '@documenso/ui/primitives/use-toast';
+
+const RESEND_CONFIRMATION_EMAIL_TIMEOUT = 20 * ONE_SECOND;
+
+export default function UnverifiedAccount() {
```

Comment:
I think it may be more optimal to have this page not be a client page, so we can immediately decrypt the search param on the server side, then check if it's an email and pass it to a client component which handles the rest. 

It'll let us:
- Display the email (if required)
- Handle invalid emails directly (error state)
- Frees up the restriction we have on the sendConfirmation endpoint where the value has to be encrypted

Thoughts? Just a suggestion

Reply:
Thanks for the review! Sounds good, but I have the following questions:
1 - when would we display the email on the `unverified page`?
2 - wouldn't invalid emails be handled in the registration process? The user is redirected directly to the `unverified page` after signing up/logging. So the user can't manually provide an invalid address. Or am I missing something?
3 - does that make any difference?

All in all, sounds good. Just the above questions for my understanding.

Reply:
#### 1 - when would we display the email on the unverified page?

It could potentially be used in the template, or in a toast when the email is resent

Example, the current template doesn't make too much sense since there's no partial access anymore

```html
<p className="text-muted-foreground mt-4">
  To gain full access to your account and unlock all its features, please confirm your email
  address by clicking on the link sent to your email address.
</p>
```

We could do something like this, which would be more informative

> An email has been sent to your email address something@something.com. Please confirm...




####  2 - wouldn't invalid emails be handled in the registration process? The user is redirected directly to the unverified page after signing up/logging. So the user can't manually provide an invalid address. Or am I missing something?

The search param is editable, so this could theoretically be changed to anything (highly unlikely, and no real reason to)

Having it done server side would allow us to check if there is even a pending user for this email, so we can display an appropriate message for that scenario as well if we wanted to

#### 3 - does that make any difference?

Nope, since I don't think this endpoint would be used elsewhere (?), it's just cleaner

Reply:
1 - hmmm, ok. I understand.

2 - it is editable indeed, but isn't it encrypted?

3 - for now, yes.

Ok then, let's make this a server component.